libs/core/src/reactor.cpp

```diff
@@ -132,39 +137,76 @@ void Reactor::Stop()
 void Reactor::StartWorker()
 {
   detailed_assert(!worker_);
+  detailed_assert(!watcher_);
 
   // signal the reactor is running
   running_ = true;
 
   // create the worker routine
   worker_ = std::make_unique<ProtectedThread>(&Reactor::Monitor, this);
+
+  // The reactor watcher determines whether executions are taking
+  // too long or are stalled.
+  watcher_ = std::make_unique<ProtectedThread>(&Reactor::ReactorWatch, this);
 }
 
 Reactor::~Reactor()
 {
-  if (worker_)
-  {
-    worker_->ApplyVoid([](auto &worker) { worker.join(); });
-    worker_.reset();
-  }
+  StopWorker();
 }
 
 void Reactor::StopWorker()
```

Comment:
A better name perhaps, since it stops both?

Reply:
renamed them to StopWorkerAndWatcher etc.